File: src/main/java/burrow/furniture/standard/ConfigGetCommand.java
package burrow.furniture.standard;

import burrow.core.chamber.ChamberInitializationException;
import burrow.core.command.Command;
import burrow.core.command.CommandContext;
import burrow.core.command.CommandType;
import burrow.core.common.ColorUtility;
import burrow.core.common.Values;
import picocli.CommandLine;

@CommandLine.Command(
    name = "c.get",
    description = "Retrieve the value of a configuration entry.")
@CommandType(StandardFurniture.COMMAND_TYPE)
public class ConfigGetCommand extends Command {
    @CommandLine.Parameters(index = "0", description = "The key of the entry to retrieve.")
    private String key;

    public ConfigGetCommand(final CommandContext commandContext) {
        super(commandContext);
    }

    @Override
    public Integer call() throws ChamberInitializationException {
        final var standardFurniture = use(StandardFurniture.class);
        final var value = standardFurniture.retrieveConfigItem(key);
        if (value == null) {
            buffer.append(ColorUtility.render(Values.NULL, ColorUtility.Type.NULL));
        } else {
            buffer.append(ColorUtility.render(value, ColorUtility.Type.VALUE));
        }

        return CommandLine.ExitCode.OK;
    }
}

File: src/main/kotlin/carton/core/command/ConfigGetCommand.kt
package burrow.carton.core.command

import burrow.kernel.terminal.*

@BurrowCommand(
    name = "config.get",
    description = ["Get the config value associated with a specific key."]
)
class ConfigGetCommand(data: CommandData) : Command(data) {
    @Parameters(
        index = "0",
        description = ["The key associated with the value"]
    )
    private var key = ""

    override fun call(): Int {
        val value = config.entries[key]
        if (value == null) {
            stderr.println("Key is not allowed: $key")
            return ExitCode.USAGE
        }

        stdout.println(value)

        return ExitCode.OK
    }
}